vercel/vercel
# [cli] implements vc deploy --logs and vc inspect --logs

### 🧐 What's in there?

This is the first part of distinguishing build logs from runtime logs.
The proposal is to provide:
- `vc deploy --logs`: in addition to deploying the application, it should tail and display the **build** logs until the build finishes.
- `vc inspect --logs`: instead of retuning a summary of the builds, it reads the **build** logs
- `vc inspect --logs --wait`: when the build is in progress, tails the **build** logs until the configured timeout (3m by default), or display all available logs when the build is finished.
- `vc logs`: read the **runtime** logs (not part of this PR, breaking change)

### 🧪 How to test?

Given a local application location in `xyz` folder, already linked to a vercel project:
- `pnpm -F vercel dev deploy --cwd xyz --logs` to deploy and see build logs
- `pnpm -F vercel dev deploy --cwd xyz` to deploy and see spinners instead
- `pnpm -F vercel dev inspect --cwd xyz dpl_ID/url --logs` to see build logs of an existing deployment
- `pnpm -F vercel dev inspect --cwd xyz dpl_ID/url` to see an existing deployment details
- `pnpm -F vercel dev inspect --cwd xyz dpl_ID/url --logs --wait` to see build logs of an existing deployment (better if in progress)
- `pnpm -F vercel dev inspect --cwd xyz dpl_ID/url --logs --wait --timeout 5s` to see build logs of an existing deployment (better if in progress) and bail after 5 seconds


Unit tests added:
- inspect command: `pnpm -F vercel exec vitest run /inspect.test`
- deploy command: `pnpm -F vercel exec vitest run /deploy.test`

### ❗ Notes to reviewers

It's better to [omit whitespace changes](https://github.com/vercel/vercel/pull/11672/files?diff=split&w=1) when reviewing `packages/cli/src/util/events.ts` since most of the code was enclosed in a try-catch.
